numpy/core/arrayprint.py

```diff
@@ -632,71 +632,79 @@ def _formatArray(a, format_function, rank, max_line_len, next_line_prefix,
     2. Summarized output
 
     """
-    if rank == 0:
```

Comment:
The `rank` variable was a workaround for dealing with `dtype=object` arrays, where `arr[i].ndim > arr.ndim` is possible, and an external counter is needed. By indexing the array only once, we avoid that confusion.